src/components/views/right_panel/RoomHeaderButtons.tsx

```diff
@@ -38,9 +42,35 @@ const ROOM_INFO_PHASES = [
     RightPanelPhases.Room3pidMemberInfo,
 ];
 
+const PinnedMessagesHeaderButton = ({ room, isHighlighted, onClick }) => {
+    const pinningEnabled = useSettingValue("feature_pinning");
+    const pinnedEvents = usePinnedEvents(pinningEnabled && room);
+    const readPinnedEvents = useReadPinnedEvents(pinningEnabled && room);
+    if (!pinningEnabled) return null;
```

Comment:
we can probably short circuit this by not reading pinned events if the setting isn't enabled. ie: move new lines 47 and 48 to below this condition:
```suggestion
    const pinningEnabled = useSettingValue("feature_pinning");
    if (!pinningEnabled) return null;

    const pinnedEvents = usePinnedEvents(pinningEnabled && room);
    const readPinnedEvents = useReadPinnedEvents(pinningEnabled && room);
```

Reply:
Hooks cannot be conditional.
But the two hooks here are effectively no-opd if you pass them a falsey as we are here.

The alternative would be tracking feature_pinning in RoomHeaderButtons but that's more lines of code due to the class setup than a `useSettingValue` hook call

Reply:
well that's not very fun :( 

This is fine, then.